backend/entityservice/views/project.py

```diff
@@ -121,57 +121,69 @@ def project_clks_post(project_id):
     with DBConn() as conn:
         dp_id = db.get_dataprovider_id(conn, token)
         project_encoding_size = db.get_project_schema_encoding_size(conn, project_id)
+        upload_state_updated = db.get_and_set_dataprovider_upload_state_in_progress(conn, dp_id)
```

Comment:
this is a bit cryptic. The function `get_and_set_dataprovider_upload_state_in_progress` mixes two tasks. 
How about if we split this to make it clearer:
```
if not db.dataprovider_allowed_to_upload(conn, project_id, dp_id, ...):
  return safe_fail_request(403, "This token has already been used to upload clks.")
db.set_dp_upload_state_in_progress(...)
...
```

Reply:
The main point is to do both actions at once not to have race conditions. Otherwise we would first assess it, and then setting it which should return an error if already set. I thus rename the method `is_dataprovider_allowed_to_upload_and_lock`.

Reply:
I see...